kimhaejung/services/userService.js

```diff
@@ -0,0 +1,48 @@
+const userDao = require("../models/userDao");
+
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+
+const signUp = async (name, email, password, profileImage) => {
+  const saltRounds = 10;
+  const makeHash = async (password, saltRounds) => {
+    return await bcrypt.hash(password, saltRounds);
+  };
+
+  const hashedPassword = await makeHash(password, saltRounds);
+  const createUser = await userDao.createUser(
+    name,
+    email,
+    hashedPassword,
+    profileImage
+  );
+  return createUser;
+};
+
+const signIn = async (email, password) => {
+  const user = await userDao.findUser(email);
+  console.log(user.password);
+
+  if (!user) {
+    const err = new Error("INVALID_EMAIL");
+    err.statusCode = 400;
+    throw err;
+  }
+  console.log(password, user.password);
+  const isMatched = await bcrypt.compare(password, user.password);
+
+  if (!isMatched) {
+    const err = new Error("WRONG_PASSWORD");
+    err.statusCode = 400;
+    throw err;
+  }
```

Comment:
로그인 실패시 어떤 정보가 틀렸는지 모르게 하는 것이 좋습니다.
그리고 상태 코드는 401이 적절 할 것 같아요.
```suggestion
  if (!isMatched) {
    const err = new Error("INVALID_USER");
    err.statusCode = 401;
    throw err;
  }
```